os/board/common/partitions.c

```diff
@@ -211,13 +260,12 @@ int configure_mtd_partitions(struct mtd_dev_s *mtd, struct partition_data_s *par
 
 		type_specific_initialize(minor, mtd_part, partno, types);
 
-#ifdef CONFIG_MTD_PARTITION_NAMES
-		configure_partition_name(mtd_part, (const char **)&names, &index, part_name);
-#ifdef CONFIG_BINARY_MANAGER
 		if (!strncmp(types, "kernel,", 7)) {
-			binary_manager_register_kpart(partno, partsize);
+			partitions_register_kernel(partno, partsize);
```

Comment:
If this API fails, it just returns with invalid part info logs (no return value)
Do we not need to take any further action here for kernel configuration failure and move on to next partition ?